### PR TITLE
Fix/functions

### LINKED ISSUE #41: Functions Library
## User Story
**As a**: developer

**I want to**: have access the a variety of re-useable functions across all our projects

**Because**: we have several copies these functions living in various projects which don't have tests and exist in various states of development

## Additional Context
_Add any context, screenshots, designs, links, etc. about the feature request here._
https://github.com/bonlineza/luxity-sales/pull/151#pullrequestreview-289992569
note that some functions are project specific - please list functions that should be migrated here in the Acceptance Criteria

## Acceptance Criteria
_List of conditions that mark the story as complete_
- [x] move un-opinionated and shared functions to library
- [x] tests to be written for EACH function added to the library 
- [x] documentation for each function - jsdoc (@return, @params, @description)
- [x] smoothScrollTo
- [x] constrainable
- [ ] etc, list them out here first...

### Edge cases
_List of edge cases to be considered when marking the story as complete_
- [ ] project usages must be updated to use the new library location

----

## Technical Description (not required initially)
_A technical description of the problem, this should be completed by the engineering manager/developer_
function/files under the `functions` folders in projects should be migrated to the library (if they are seen as needed in many projects)

### PR DESCRIPTION
**Pull request by sscholle**
## Proposed changes
- Add constrainable and smoothScrollTo functions
- Export Functions
- Add tests for functions
- Add documentation for functions

## Types of changes

What type of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] New feature
- [ ] Bugfix
- [ ] Quality of Life

## Checklist

_Put an `x` in the boxes that apply._

- [x] Linter passes (`yarn lint`)
- [x] Unit Test pass (`yarn jest`)
- [ ] I have added at least two useful tests for new component (if applicable)
- [ ] I have added at least one Storybook story (if applicable)

## Relevant Issues

_List Issues below in format `#<ISSUE-NUMBER> [Full|Partial]`_
- Resolves #41

### CODE REVIEW

File: src/util/getToken.js [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
 
-  const token = window.localStorage.getItem('_tk');
+  const token = window.localStorage.getItem(key);
   if (token === null) return null;
 
-  return withBearer ? token : token.substr(7);
+  if (withBearer) {
+    return token;
+  }
+  return token.substr(7);
```
Comment (robguy21) [inline src/util/getToken.js]: this was changed in allianz-

```js
export default function getToken(withBearer = false) {
  if (typeof window === 'undefined') return null;

  const token = window.localStorage.getItem('_tk');
  if (token === null) return null;

  return withBearer ? token : token.substr(7);
}
```

Comment (robguy21) [reply src/util/getToken.js]: the above won't error out if `token` is some non string non null value (it happened)

File: README.md [hunk 2]
```diff
@@ -128,28 +132,42 @@ We can use this design to "override" certain packages.
...
 cd /path/to/project/resources/assets/js
-
 mkdir node_modules
-
 cd node_modules
-
 git clone git@github.com:{author}/b-lib --branch {branch-name}
 ```
 
+## Changelog
```
Comment (robguy21) [inline README.md]: let's rather use release notes for this- I don't think this is a sustainable way of doing releases.

My suggestion is to use squash commits + annotated tags + `git notes` + `git whatchanged {old_version}...{new version}` when we do a release

We can turn this into an issue for future discussion

Comment (sscholle) [reply README.md]: added issue: https://github.com/bonlineza/b-lib/issues/59

Comment (robguy21) [reply README.md]: Resolving as issue exists

File: src/util/getToken.js (hunk 1 above)
Comment (sscholle) [reply src/util/getToken.js]: updated token check

**Pull request merged**